pkg/persistence/redis/redisclient.go

```diff
@@ -0,0 +1,160 @@
+package redis
+
+import (
+"errors"
+"fmt"
+"net/url"
+"strconv"
+"strings"
+"sync"
+"time"
+
+"github.com/FZambia/sentinel"
+"github.com/gomodule/redigo/redis"
+)
+
+var knownPool sync.Map
+var m sync.Mutex
+
+// RedisPoolParam ...
+type RedisPoolParam struct {
+	PoolMaxIdle     int
+	PoolMaxActive   int
+	PoolIdleTimeout time.Duration
+
+	DialConnectionTimeout time.Duration
+	DialReadTimeout       time.Duration
+	DialWriteTimeout      time.Duration
+}
+
+// GetRedisPool get a named redis pool
+// supported schema
+// redis://user:pass@redis_host:port/db
+// redis+sentinel://user:pass@redis_sentinel1:port1,redis_sentinel2:port2/monitor_name/db
+func GetRedisPool(name string, rawurl string, param *RedisPoolParam) (*redis.Pool, error) {
+	if p, ok := knownPool.Load(name); ok {
+		return p.(*redis.Pool), nil
+	}
+	m.Lock()
+	defer m.Unlock()
+	// load again in case multi threads
```

Comment:
Apparently the `GetRedisPool` method is called once, in the main function where we construct the `Store` by calling `redis.NewStore()`. Hence, I'm wondering if we really need all that code to guard the pool from being accessed by multiple goroutines?

Reply:
Because sentinel actually need using a pool to keep it working, each time we get a connection, sentinel will return if need to change the master address

Reply:
Got it. I think we're good with just a single instance of the pool.